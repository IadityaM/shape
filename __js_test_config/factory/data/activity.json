{"data":{"id":"1365201","type":"activities","attributes":{"action":null,"created_at":"2019-12-16T23:14:35.973Z","content":null,"actor_id":22548,"target_type":"Collection","target_id":762911,"source_name":""},"relationships":{"actor":{"meta":{"included":false}},"subject_users":{"meta":{"included":false}},"subject_groups":{"meta":{"included":false}},"target":{"meta":{"included":false}}}}}