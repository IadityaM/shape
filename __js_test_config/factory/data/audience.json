{"data":{"id":"104","type":"audiences","attributes":{"name":"Natural Armor","global_default":null,"age_list":[],"children_age_list":[],"country_list":[],"education_level_list":[],"gender_list":[],"adopter_type_list":[],"interest_list":[],"publication_list":[],"min_price_per_response":4.0,"order":null,"global":true}}}