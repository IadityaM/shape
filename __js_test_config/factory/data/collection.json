{"data":{"id":"762903","type":"collections","attributes":{"class_type":"Collection","name":"Extended","created_at":"2019-12-16T23:14:33.643Z","updated_at":"2019-12-16T23:14:33.643Z","master_template":false,"template_id":null,"submission_box_type":null,"submission_box_id":null,"submission_template_id":null,"test_status":null,"collection_to_test_id":null,"hide_submissions":false,"submissions_enabled":true,"anyone_can_view":false,"anyone_can_join":false,"cover_type":"cover_type_default","archived":false,"unresolved_count":0,"last_unresolved_comment_id":null,"test_show_media":true,"idea_id":null,"search_term":null,"organization_id":"5352","joinable_group_id":null,"default_group_id":null,"system_required":false,"tag_list":[],"inherited_tag_list":[],"cover":{"text":"","hardcoded_subtitle":"","subtitle_hidden":false},"test_scores":{},"breadcrumb":[{"type":"collections","id":"762903","name":"Extended","can_edit":false}],"processing_status":null,"collection_card_count":0,"can_view":false,"can_edit":false,"can_edit_content":false,"is_private":false,"is_org_template_collection":false,"is_profile_template":false,"is_profile_collection":false,"pinned_and_locked":false,"is_submission_box_template":false,"is_submission_box_template_test":false,"is_inside_hidden_submission_box":false,"is_inside_a_submission":false,"is_subtemplate_or_instance":false,"show_language_selector":false,"template_num_instances":0,"awaiting_updates":null,"num_survey_responses":0,"max_row_index":null,"max_col_index":null,"frontend_url":"http://localhost/","common_viewable":null,"is_test_locked":null,"has_link_sharing":null,"is_restorable":false,"serializer":"SerializableCollection"},"relationships":{"roles":{"meta":{"included":false}},"collection_cover_items":{"meta":{"included":false}},"parent_collection_card":{"meta":{"included":false}},"parent":{"meta":{"included":false}},"live_test_collection":{"meta":{"included":false}},"submissions_collection":{"meta":{"included":false}},"submission_template":{"meta":{"included":false}},"collection_to_test":{"meta":{"included":false}},"organization":{"meta":{"included":false}},"created_by":{"meta":{"included":false}},"test_audiences":{"meta":{"included":false}},"test_results_collection":{"meta":{"included":false}},"collection_filters":{"meta":{"included":false}},"restorable_parent":{"meta":{"included":false}}}}}