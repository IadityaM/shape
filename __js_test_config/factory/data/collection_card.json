{"data":{"id":"2877260","type":"collection_cards","attributes":{"class_type":"CollectionCard::Primary","order":0,"width":1,"height":1,"parent_id":762904,"pinned":false,"image_contain":false,"card_question_type":null,"is_cover":false,"filter":"transparent_gray","hidden":false,"show_replace":true,"updated_at":"2019-12-16T23:14:34.478Z","col":1,"row":1,"section_type":null,"idea_id":null,"link":false,"pinned_and_locked":false,"can_edit_parent":false,"is_master_template_card":false},"relationships":{"item":{"meta":{"included":false}},"collection":{"meta":{"included":false}},"parent":{"meta":{"included":false}},"record":{"meta":{"included":false}},"templated_from":{"meta":{"included":false}}}}}