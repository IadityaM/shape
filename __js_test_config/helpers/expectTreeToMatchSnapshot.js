import toJson from 'enzyme-to-json'

const expectTreeToMatchSnapshot = tree => {
  expect(toJson(tree)).toMatchSnapshot()
}

export default expectTreeToMatchSnapshot
