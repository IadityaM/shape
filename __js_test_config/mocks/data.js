export const fakeTextItemAttrs = {
  id: 1,
  type: 'Item::TextItem',
  name: 'My Cool Item',
  text_data: 'This is the content for the item and it contains multiple sentences. Like this one.',
  breadcrumb: [['collections', 1, 'Some collection'], ['items', 1, 'my item']],
  parentPath: '/',
}
export const fakeTextItem = {
  ...fakeTextItemAttrs,
  rawAttributes: jest.fn().mockReturnValue(fakeTextItemAttrs),
  getRecordType: jest.fn().mockReturnValue('items'),
  toJS: jest.fn().mockReturnValue(fakeTextItemAttrs),
}
export const fakeVideoItemAttrs = {
  id: 2,
  type: 'Item::VideoItem',
  name: 'Crazy Honey Badger',
  url: 'https://www.youtube.com/watch?v=4r7wHMg5Yjg'
}
export const fakeVideoItem = {
  ...fakeVideoItemAttrs,
  rawAttributes: jest.fn().mockReturnValue(fakeTextItemAttrs),
  getRecordType: jest.fn().mockReturnValue('items'),
}
export const fakeImageItemAttrs = {
  id: 3,
  type: 'Item::ImageItem',
  name: 'Earth from Space',
  filestack_file: {
    id: 1,
    url: 'https://www.nasa.gov/sites/default/files/styles/full_width_feature/public/thumbnails/image/iss052e023801_0.jpg',
  }
}
export const fakeImageItem = {
  ...fakeImageItemAttrs,
  rawAttributes: jest.fn().mockReturnValue(fakeTextItemAttrs),
  getRecordType: jest.fn().mockReturnValue('items'),
}
export const fakeItemCard = {
  id: 10,
  order: 0,
  height: 1,
  width: 1,
  record: fakeTextItem,
  item: fakeTextItem,
}
export const fakeCollectionCard = {
  id: 11,
  order: 1,
  height: 1,
  width: 1,
  record: fakeCollection,
  item: fakeCollection,
  reference: false,
}
export const fakeCollection = {
  id: 1,
  name: 'My Workspace X',
  type: 'Collection',
  breadcrumb: [],
  collection_cards: [
    fakeItemCard, fakeItemCard, fakeItemCard
  ]
}
export const fakePosition = {
  xPos: 0,
  yPos: 0,
  height: 1,
  width: 1,
}
