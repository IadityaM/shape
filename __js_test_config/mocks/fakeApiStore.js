const fakeApiStore = ({ findResult = '', findAllResult = [], requestResult = '' } = {}) => {
  return {
    add: jest.fn().mockName('add'),
    removeAll: jest.fn().mockName('removeAll'),
    find: jest.fn()
      .mockReturnValue(findResult),
    findAll: jest.fn()
      .mockReturnValue(findAllResult),
    request: jest.fn()
      .mockReturnValue(Promise.resolve(requestResult))
      .mockName('request'),
    fetch: jest.fn()
      .mockReturnValue(Promise.resolve())
      .mockName('fetch'),
    sync: jest.fn().mockName('sync'),
    setCurrentUserId: jest.fn(),
    currentUserId: 1,
    currentUser: {
      id: 1,
      current_user_collection_id: 99,
      current_organization: {
        name: 'test org'
      },
      name: 'Johnny Appleseed',
      pic_url_square: 'https://s3.amazonaws.com/pic.png',
      groups: [],
      terms_accepted: true,
    },
    collections: [],
    items: [],
  }
}

export default fakeApiStore
