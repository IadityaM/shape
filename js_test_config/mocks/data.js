export const fakeItemAttrs = {
  id: 1,
  name: 'My Cool Item',
}
export const fakeItem = {
  ...fakeItemAttrs,
  rawAttributes: jest.fn().mockReturnValue(fakeItemAttrs),
  getRecordType: jest.fn().mockReturnValue('items'),
}
export const fakeCard = {
  id: 10,
  order: 0,
  height: 1,
  width: 1,
  record: fakeItem,
  item: fakeItem,
}
export const fakeCollection = {
  id: 1,
  name: 'My Workspace X',
  breadcrumb: [],
  collection_cards: [
    fakeCard, fakeCard, fakeCard
  ]
}
export const fakePosition = {
  xPos: 0,
  yPos: 0,
  height: 1,
  width: 1,
}
