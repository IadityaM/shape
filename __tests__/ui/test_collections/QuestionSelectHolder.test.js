import QuestionSelectHolder from '~/ui/test_collections/QuestionSelectHolder'
import { fakeCollection } from '#/mocks/data'
import fakeApiStore from '#/mocks/fakeApiStore'
import expectTreeToMatchSnapshot from '#/helpers/expectTreeToMatchSnapshot'

import v from '~/utils/variables'

let wrapper, props, instance, component
describe('TestDesigner', () => {
  beforeEach(() => {
    props = {
      collection: fakeCollection,
    }
    // very basic way to turn fakeCollection into a "test collection"
    props.collection.collection_cards[0].card_question_type = 'question_useful'
    props.collection.apiStore = fakeApiStore({
      requestResult: { data: { id: 99, name: 'Parent Collection' } },
    })
    wrapper = shallow(<QuestionSelectHolder {...props} />)
  })

  it('renders snapshot', () => {
    expectTreeToMatchSnapshot(wrapper)
  })
})
