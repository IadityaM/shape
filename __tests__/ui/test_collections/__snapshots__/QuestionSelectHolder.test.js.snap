// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestDesigner renders snapshot 1`] = `
<styled.div>
  <NumberListText>
    2
    .
  </NumberListText>
  <StyledSelect
    classes={
      Object {
        "root": "select fixedWidth",
        "select": "",
        "selectMenu": "selectMenu bottomPadded",
      }
    }
    disabled={false}
    displayEmpty={true}
    name="role"
    value="question_useful"
  >
    <StyledSelectOption
      classes={
        Object {
          "root": "grayedOut",
          "selected": "selected",
        }
      }
      disabled={true}
      key=""
      value=""
    >
      <span
        data-cy="QuestionSelectOption-select-question-type"
      >
        select question type
      </span>
    </StyledSelectOption>
    <StyledSelectOption
      classes={
        Object {
          "root": "category",
          "selected": "selected",
        }
      }
      disabled={true}
      key=""
      value=""
    >
      <span
        data-cy="QuestionSelectOption-idea-content"
      >
        Idea Content
      </span>
    </StyledSelectOption>
    <StyledSelectOption
      classes={
        Object {
          "root": "selectOption",
          "selected": "selected",
        }
      }
      disabled={false}
      key="question_description"
      value="question_description"
    >
      <span
        data-cy="QuestionSelectOption-description"
      >
        Description
      </span>
    </StyledSelectOption>
    <StyledSelectOption
      classes={
        Object {
          "root": "selectOption",
          "selected": "selected",
        }
      }
      disabled={false}
      key="question_idea"
      value="question_idea"
    >
      <span
        data-cy="QuestionSelectOption-idea"
      >
        Idea
      </span>
    </StyledSelectOption>
    <StyledSelectOption
      classes={
        Object {
          "root": "selectOption",
          "selected": "selected",
        }
      }
      disabled={false}
      key="question_idea_placeholder"
      value="question_idea_placeholder"
    >
      <span
        data-cy="QuestionSelectOption-idea-pl"
      >
        Idea (pl)
      </span>
    </StyledSelectOption>
    <StyledSelectOption
      classes={
        Object {
          "root": "category",
          "selected": "selected",
        }
      }
      disabled={true}
      key=""
      value=""
    >
      <span
        data-cy="QuestionSelectOption-scaled-rating"
      >
        Scaled Rating
      </span>
    </StyledSelectOption>
    <StyledSelectOption
      classes={
        Object {
          "root": "selectOption",
          "selected": "selected",
        }
      }
      disabled={false}
      key="question_clarity"
      value="question_clarity"
    >
      <span
        data-cy="QuestionSelectOption-clear"
      >
        Clear
      </span>
    </StyledSelectOption>
    <StyledSelectOption
      classes={
        Object {
          "root": "selectOption",
          "selected": "selected",
        }
      }
      disabled={false}
      key="question_different"
      value="question_different"
    >
      <span
        data-cy="QuestionSelectOption-different"
      >
        Different
      </span>
    </StyledSelectOption>
    <StyledSelectOption
      classes={
        Object {
          "root": "selectOption",
          "selected": "selected",
        }
      }
      disabled={false}
      key="question_excitement"
      value="question_excitement"
    >
      <span
        data-cy="QuestionSelectOption-exciting"
      >
        Exciting
      </span>
    </StyledSelectOption>
    <StyledSelectOption
      classes={
        Object {
          "root": "selectOption",
          "selected": "selected",
        }
      }
      disabled={false}
      key="question_useful"
      value="question_useful"
    >
      <span
        data-cy="QuestionSelectOption-useful"
      >
        Useful
      </span>
    </StyledSelectOption>
    <StyledSelectOption
      classes={
        Object {
          "root": "category",
          "selected": "selected",
        }
      }
      disabled={true}
      key=""
      value=""
    >
      <span
        data-cy="QuestionSelectOption-customizable"
      >
        Customizable
      </span>
    </StyledSelectOption>
    <StyledSelectOption
      classes={
        Object {
          "root": "selectOption",
          "selected": "selected",
        }
      }
      disabled={false}
      key="question_open"
      value="question_open"
    >
      <span
        data-cy="QuestionSelectOption-open-response"
      >
        Open Response
      </span>
    </StyledSelectOption>
  </StyledSelect>
  <styled.button
    onClick={[Function]}
  >
    <TrashIcon />
  </styled.button>
  <div
    style={
      Object {
        "color": "#c6c1bf",
      }
    }
  />
</styled.div>
`;
